backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interim/bpartner/BPartnerInterimContractService.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * #%L
+ * de.metas.contracts
+ * %%
+ * Copyright (C) 2023 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.contracts.modular.interim.bpartner;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import de.metas.bpartner.BPartnerId;
+import de.metas.contracts.FlatrateTermId;
+import de.metas.contracts.FlatrateTermRequest.ModularFlatrateTermRequest;
+import de.metas.contracts.IFlatrateBL;
+import de.metas.contracts.flatrate.TypeConditions;
+import de.metas.contracts.modular.settings.ModularContractSettings;
+import de.metas.contracts.modular.settings.ModularContractSettingsDAO;
+import de.metas.util.Check;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.exceptions.AdempiereException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class BPartnerInterimContractService
+{
+	private final IFlatrateBL flatrateBL = Services.get(IFlatrateBL.class);
+
+	private final BPartnerInterimContractRepo bPartnerInterimContractRepo;
+	private final ModularContractSettingsDAO modularContractSettingsDAO;
+
+	public BPartnerInterimContractService(
+			@NonNull final BPartnerInterimContractRepo bPartnerInterimContractRepo,
+			@NonNull final ModularContractSettingsDAO modularContractSettingsDAO)
+	{
+		this.bPartnerInterimContractRepo = bPartnerInterimContractRepo;
+		this.modularContractSettingsDAO = modularContractSettingsDAO;
+	}
+
+	public void upsert(@NonNull final BPartnerInterimContractUpsertRequest request)
```

Comment:
this method feels quite inefficient with a lot of loads.....
low prior, tackle it only if u have time